Extension/src/LanguageServer/configurations.ts

```diff
@@ -225,6 +229,35 @@ export class CppProperties {
         }
     }
 
+    private async buildVcpkgIncludePath() {
+        const vcpkgConfigPath:string = "c:/users/jimgries/AppData/Local/vcpkg/vcpkg.path.txt";
+
+        try {
+            // Check for vcpkg instance and include relevent paths if found.
+            if (await util.checkFileExists(util.getVcpkgHomePath())) {
+                let vcpkgRoot: string = await util.readFileText(vcpkgConfigPath);
+                let vcpkgInstallPath: string = path.join(vcpkgRoot.slice(0, vcpkgRoot.lastIndexOf("\r\n")), "/vcpkg/installed");
+                if (await util.checkDirectoryExists(vcpkgInstallPath)) {
+                    fs.readdir(vcpkgInstallPath, (err, list) => {
+                        // For every *directory* in the list (non-recursive)
+                        list.forEach((entry) => {
+                            if (entry != "vcpkg") {
+                               let pathToCheck: string = path.join(vcpkgInstallPath, entry);
+                               if (fs.existsSync(pathToCheck)) {
+                                   var s = path.join(pathToCheck,"include");
+                                   this.vcpkgIncludes.push(s);
+                               }
+                            }
+                        });
+                        this.vcpkgPathReady = true;
+                        this.handleConfigurationChange();
+                    })
+                };
+            }
+        }
+        catch(error) {}
```

Comment:
catch should set this.vcpkgPathReady = true and call this.handleConfigurationChange() so that we don't get stuck forever waiting for data that's never coming.

Reply:
Ah yeah. But a finally would be better for this, no? We need to call these two operations no matter what.